pkg/util/olm/operator.go

```diff
@@ -343,17 +345,31 @@ func fillDefaults(o Options) Options {
 	if o.Channel == "" {
 		o.Channel = DefaultChannel
 	}
-	if o.Source == "" {
-		o.Source = DefaultSource
-	}
-	if o.SourceNamespace == "" {
-		o.SourceNamespace = DefaultSourceNamespace
-	}
 	if o.StartingCSV == "" {
 		o.StartingCSV = DefaultStartingCSV
 	}
-	if o.GlobalNamespace == "" {
-		o.GlobalNamespace = DefaultGlobalNamespace
+	if isOCP, err := openshift.IsOpenShift(client); isOCP && err != nil {
```

Comment:
I'm not sure I understand the intent of `err != nil`. It seems the error should be returned instead.

Reply:
I did not want to change the function signature, neither ignore the `err` return value. The idea is that if an error is returned, then we always default to the community configuration. Having a look at the `openshift.IsOpenShift` I don't expect any kind of error though. And if there is any cluster error, then we should not reach this point.

Reply:
Is there any reason why you don't want to change the function signature? I'd rather keep it idiomatic and fail fast, instead of introducing that pattern that is hard to read for the external developer.

Reply:
I originally though it involved bubbling up the error into too many levels. Turned out it was not the case as we already manage an error condition in the same calling method. Resolved via 58dc6351eebb9a59f58b42adbe16a04c9e8aef4c